h/static/scripts/account/test/auth-controller-test.coffee

```diff
@@ -18,15 +18,36 @@ class MockSession
     $promise:
       finally: sandbox.stub()
 
-mockFlash = info: sandbox.spy()
-mockFormRespond = sandbox.spy()
-
-describe 'h:AuthController', ->
+# Return an AuthController instance and associated objects for tests to use.
+_controller = (mocksession) ->
+  controller = null
   $scope = null
   $timeout = null
-  auth = null
   session = null
-  $controller = null
+  mockFlash = info: sandbox.spy()
+  mockFormRespond = sandbox.spy()
+  inject(($controller, $rootScope) ->
```

Comment:
Wouldn't it be simpler to use `inject` to capture `$controller` and `$rootScope` only, and then you don't need to close over `controller`, you just have:

```coffeescript
$controller = null
$rootScope = null

inject (_$controller_, _$rootScope_) -> 
  $controller = _$controller_
  $rootScope = _$rootScope_

locals = {
  $scope: $rootScope.$new()
  $timeout: sandbox.spy()
  flash: mockFlash
  session: mocksession or new MockSession()
  ...
}

auth = $controller('AuthController', locals)
locals.auth = auth

return locals
```